### PR TITLE
Move services from Save and COSV into a common module

### PR DESCRIPTION
**Pull request by Cheshiriks**

### CODE REVIEW

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/service/CommentService.kt [hunk 1]
```diff
@@ -31,8 +31,8 @@ class CommentService(
      */
     @Transactional
     fun saveComment(comment: CommentDto, authentication: Authentication) {
-        val userId = authentication.userId()
-        val user = userRepository.getByIdOrNotFound(userId)
+        val userName = authentication.username()
```
Comment (kgevorkyan) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/service/CommentService.kt:34]: what's the point to change id to name?

Comment (Cheshiriks) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/service/CommentService.kt:34]: In the common module we do not have access to methods from the authentication service, so we cannot get the ID (authentication.userId()) and have to use the name

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * @param name name of user
+     * @param authentication
+     * @return UserSaveStatus
+     */
+    @Transactional
+    fun deleteUser(
+        name: String,
+        authentication: Authentication,
+    ): UserSaveStatus {
```
Comment (akuleshov7) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt:52]: Let's return it back to common, but the path to avatar need to be propagated into the common application

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            this.gitHub = null
+            this.linkedin = null
+            this.location = null
+        })
+
+        val avatarKey = AvatarKey(
+            AvatarType.USER,
+            name,
+        )
+        avatarStorage.delete(avatarKey)
```
Comment (akuleshov7) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt:75]: Should it be moved to common?

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+) {
+    /**
+     * @param name name of user
+     * @param authentication
+     * @return UserSaveStatus
+     */
+    @Transactional
+    fun deleteUser(
+        name: String,
+        authentication: Authentication,
```
Comment (akuleshov7) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt:51]: configProperties: S3OperationsProperties.Provider,

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+@Service
+@Suppress("LongParameterList")
+@Primary
+class CosvUserService(
+    private val userRepository: UserRepository,
+    private val applicationEventPublisher: ApplicationEventPublisher,
+    private val originalLoginRepository: OriginalLoginRepository,
+    private val lnkUserOrganizationRepository: LnkUserOrganizationRepository,
+    private val lnkUserProjectRepository: LnkUserProjectRepository,
+    private val avatarStorage: AvatarStorage,
```
Comment (akuleshov7) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt:36]: change avatarStorage: configProperties: S3OperationsProperties.Provider,

**Review by akuleshov7 — APPROVED** [review]

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt]: Moved it

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt]: Moved it

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt (hunk 5 above)
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt]: Changed it

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/SaveApplication.kt [hunk 6]
```diff
@@ -10,7 +10,7 @@ import org.springframework.context.annotation.Import
 /**
  * An entrypoint for spring for save-backend
  */
-@SpringBootApplication(scanBasePackages = ["com.saveourtool.save.backend", "com.saveourtool.save.repository"])
+@SpringBootApplication(scanBasePackages = ["com.saveourtool.save.backend", "com.saveourtool.save"])
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/SaveApplication.kt:13]: looks like the problem that `com.saveourtool.save` is subset of `com.saveourtool.save.backend`.
It does work if packages specify explicitly:
```kotlin
@SpringBootApplication(scanBasePackages = [
    "com.saveourtool.save.backend",
    "com.saveourtool.save.service",
    "com.saveourtool.save.security",
    "com.saveourtool.save.storage",
    "com.saveourtool.save.utils",
])
```

**Pull request merged**